folding-schemes/src/folding/nova/mod.rs

```diff
@@ -677,7 +694,7 @@ where
         // set values for next iteration
         self.i += C1::ScalarField::one();
         self.z_i = z_i1;
-        self.w_i = Witness::<C1>::new(w_i1, self.r1cs.A.n_rows);
+        self.w_i = Witness::<C1>::new(w_i1, self.w_i.rW, self.r1cs.A.n_rows, self.w_i.rE);
```

Comment:
does this mean that `u_i.r_W` & `u_i.r_E` are propagated across all the folding steps (ie. $u_{i+1}.r_W==u_i.r_W$)? meaning for example that the `u_i.r_W` at step i=1 will be the same as in i=100?

Reply:
Maybe it could be done as it is done in hypernova, not reusing the blinding values across all the instances, but using new randomness for each new instance: [https://github.com/privacy-scaling-explorations/sonobe/folding-schemes/src/folding/hypernova/mod.rs#L765](https://github.com/privacy-scaling-explorations/sonobe/blob/6248a90e89a6890af6603138586aa3e1fb683965/folding-schemes/src/folding/hypernova/mod.rs#L765)

Reply:
That is the case, I was not too sure whether they should be re-used or not, but that should be better this way. [a06461f](https://github.com/privacy-scaling-explorations/sonobe/pull/129/commits/a06461f86ce97ca7250f60760c257f6e4822ce2a)